wallet/src/de/schildbach/wallet/data/ExchangeRatesProvider.java

```diff
@@ -209,7 +209,10 @@ public static ExchangeRate getExchangeRate(final Cursor cursor) {
                 cursor.getLong(cursor.getColumnIndexOrThrow(ExchangeRatesProvider.KEY_RATE_FIAT)));
         final String source = cursor.getString(cursor.getColumnIndexOrThrow(ExchangeRatesProvider.KEY_SOURCE));
 
-        return new ExchangeRate(new org.bitcoinj.utils.ExchangeRate(rateCoin, rateFiat), source);
+        ExchangeRate exchangeRate = new ExchangeRate(new org.bitcoinj.utils.ExchangeRate(rateCoin,
+                rateFiat), source);
+        WalletApplication.getInstance().setExchangeRate(exchangeRate);
```

Comment:
@sambarboza we shouldn't initiate the _WalletApplication.exchangeRate_ field over here, it doesn't smell good... ;)
I will take a closer look at it and try to propose something more reliable.